vtrace/platforms/gdbstub.py

```diff
@@ -892,13 +1023,22 @@ def gdbSetRegister(self, regidx, val):
         Returns:
             None
         """
-        val_str = self._encodeGDBVal(val)
+        val_str = self._encodeGDBVal(val, self._gdb_reg_fmt[regidx][1])
         cmd = b'P%.2x=%s' % (regidx, val_str)
-        self._msgExchange(cmd)
+        res = self._msgExchange(cmd)
+        if not res:
+            regname = self._gdb_reg_fmt[regidx][0]
+            logger.debug('Cannot set register %d (%s) individually' %
+                    (regidx, regname))
+            self.gdbSetRegisters({regname: val})
+        elif res[0:1] == b'E':
+            regname = self._gdb_reg_fmt[regidx][0]
+            logger.warning('Setting register %d (%s) failed with error %s' %
+                    (regidx, regname, res[1:3]))
```

Comment:
same thing here.  if we want to leave this functionality in, please make it conditional on a boolean `self._reg_tenacity` since this is *not* the spec, and these functions were written to abstract out the Spec functionalty and make it available to higher level functions.  as odd as it seems right now, this functionality may be undesireable.

Reply:
based on our discussion I am going to pull the fallback set functionality out of the top-level gdb stub class.